stix2/datastore/__init__.py

```diff
@@ -24,15 +24,35 @@ def make_id():
     return str(uuid.uuid4())
 
 
+class DataSourceError(Exception):
+    """General DataSource error instance, used primarily for wrapping
+    lower level errors
+
+    Args:
+        message (str): error message
+        root_exception (Exception): Exception instance of root exception
+            in the case that DataSourceError is wrapping a lower level or
+            other exception
+    """
+    def __init__(self, message, root_exception=None):
+        self.message = message
+        self.root_exception = root_exception
```

Comment:
Python3 already supports exception chaining via `__cause__` and `raise x from y` syntax.  Any way we could just use that in python3, instead of reinventing this wheel?

Reply:
I think we still need to support python 2 for a little while, unfortunately, but we could change this when that is no longer the case.

Reply:
Six has a [raise_from](https://pythonhosted.org/six/#six.raise_from) function... that would give code compatibility with 2 and 3, but unfortunately that would make the chaining only work under python3.  Well, I have less sympathy for python2 these days, so I went ahead and used it in the matcher.  Maybe it will push people to upgrade :)  Sigh.